flo-workflow/src/main/java/com/spotify/flo/context/MemoizingContext.java

```diff
@@ -45,6 +46,13 @@ public static EvalContext composeWith(EvalContext baseContext) {
   @SuppressWarnings("unchecked")
   @Override
   public <T> Value<T> evaluateInternal(Task<T> task, EvalContext context) {
-    return (Value<T>) ongoing.computeIfAbsent(task.id(), taskId -> super.evaluateInternal(task, context));
+    final CompletableFuture<Value<?>> f = new CompletableFuture<>();
```

Comment:
Try using `context.promise()` Instead of futures. Any existing promise in `ongoing` would then be immediately returnable.

Reply:
👍